### PR TITLE
store block responses from tendermint

### LINKED ISSUE #116: Use tendermint or LCD api for fetching blocks
### Description

We are currently using the stargate client to fetch blocks, this seems to reduce the data returned from the request.
https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/stargateclient.ts#L309

We should switch to using a TendermintClient or LCD request to provide more fields.

A requested field is `last_commit_signatures` as well as https://github.com/subquery/subql-cosmos/issues/107 maybe solved with this change

### PR DESCRIPTION
**Pull request by DeveloperInProgress**
# Description
Store block responses from tendermint directly instead of filtering out fields

Fixes #116 #107 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have tested locally
- [ ] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] My code is up to date with the base branch

### CODE REVIEW

File: packages/node/src/indexer/indexer.manager.ts [hunk 1]
```diff
@@ -163,11 +167,25 @@ export class IndexerManager
...
-    await this.indexMessage(blockContent, dataSources, getVM);
+    for (const evt of blockContent.events) {
+      await this.indexEvent(evt, dataSources, getVM);
+    }
 
-    await this.indexEvent(blockContent, dataSources, getVM);
+    for (const evt of blockContent.endBlockEvents) {
+      await this.indexEvent(evt, dataSources, getVM);
+    }
   }
```
Comment (stwiname) [inline packages/node/src/indexer/indexer.manager.ts:189]: These need to be the order they happen on chain

```
block
begin events
tx1
  msg1
    event1
    event2
  msg2
tx2
....
end events
```

File: packages/types/src/interfaces.ts [hunk 2]
```diff
@@ -34,7 +34,8 @@ export interface Store {
 }
 
 export interface CosmosBlock {
-  block: Block;
+  blockId: BlockId;
+  block: {id: string} & Block;
```
Comment (stwiname) [inline packages/types/src/interfaces.ts:38]: We need to ensure backwards compatibility. So the Tendermint block should be a superset of stargate block

Comment (DeveloperInProgress) [reply packages/types/src/interfaces.ts:38]: tendermint block is a superset of stargate block when `id` field is added to it.

```
//stargate:
export interface Block {
  /** The ID is a hash of the block header (uppercase hex) */
  readonly id: string;
  readonly header: BlockHeader;
  /** Array of raw transactions */
  readonly txs: readonly Uint8Array[];
}

//tendermint:
export interface Block {
  readonly header: Header;
  /**
   * For the block at height 1, last commit is not set.
   */
  readonly lastCommit: Commit | null;
  readonly txs: readonly Uint8Array[];
  readonly evidence: readonly Evidence[];
}
```

header is backward compatible

**Review by stwiname — COMMENTED** [review]
I'm happy with the changes here but we need to update the dictionaries to include the new events. It would be great to add some checks for whether the dictionary has these new events so we could disable it.

There is also quite a bit of commented code that can be cleaned up too

**Review by stwiname — APPROVED** [review]

**Pull request merged**